esti/migrate_test.go

```diff
@@ -0,0 +1,128 @@
+package esti
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+	"github.com/thanhpk/randstr"
+	"github.com/treeverse/lakefs/pkg/api"
+)
+
+func TestMigrate(t *testing.T) {
+	postMigrate := viper.GetViper().GetBool("post_migrate")
+
+	if postMigrate {
+		postMigrateTests(t)
+	} else {
+		preMigrateTests(t)
+	}
+}
+
+func preMigrateTests(t *testing.T) {
+	// all pre tests execution
+	t.Run("TestPreMigrateMultipart", testPreMigrateMultipart)
+
+	saveStateInLakeFS(t)
+}
+
+func postMigrateTests(t *testing.T) {
```

Comment:
Same as the comment before but for the general usecase.
If preMigrateTests failed - there's no point in running postMigrateTests.
We need to somehow indicate the pre tests failed (maybe via the state file?)

Reply:
Right - If Pre test fails -> Esti fails -> flow stops and we'll never reach the post test.